common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -243,9 +245,35 @@ class ModuleI18nService(object):
     i18n service.
 
     """
+    def __init__(self, block=None):
+        """
+        Attempt to load an XBlock-specific GNU gettext translator using the XBlock's own domain
+        translation catalog, currently expected to be found at:
+            <xblock_root>/conf/locale/<language>/LC_MESSAGES/<domain>.po|mo
+        If we can't locate the domain translation catalog then we fall-back onto
+        django.utils.translation, which will point to the system's own domain translation catalog
+        This effectively achieves translations by coincidence for an XBlock which does not provide
+        its own dedicated translation catalog along with its implementation.
+        """
+        self.translator = django.utils.translation
+        if block:
+            xblock_resource = block.unmixed_class.__module__
+            xblock_locale_dir = '/conf/locale'
+            xblock_locale_path = resource_filename(xblock_resource, xblock_locale_dir)
+            xblock_domain = 'django'
```

Comment:
It looks like the implementation of the `domain` value has evolved a bit during this PR review... Having a standard domain used by convention seems reasonable, but since XBlocks are not supposed to use django-specific APIs, shouldn't the domain be something generic like `xblock` instead of `django`? (I saw Ned asked a similar question earlier but it wasn't answered)

Also as far as I can tell this is now a fixed value for the domain, but the docstring above treats it as configurable.